client/blocks/signup-form/index.jsx

```diff
@@ -192,28 +196,19 @@ class SignupForm extends Component {
 	 * If the step is invalid because we had an error that the user exists,
 	 * we should prompt user with a request to connect his social account
 	 * to his existing WPCOM account
-	 *
-	 * @param {object} props react component props that has step info
 	 */
-	maybeRedirectToSocialConnect( props ) {
-		const userExistsError = this.getUserExistsError( props );
+	maybeRedirectToSocialConnect() {
```

Comment:
I tried to test this method, but the code path appears to be unreachable now. It looks like it was used when the user tries to signup for a regular wpcom account using an email that's already been used for a social log in account. I remember this used to only be checked by a round trip to the server, and that's when this redirect would kick in.

However, in the latest iteration of the signup form UI this case is checked without submitting the form.

I don't recommend removing the method in this commit, just pointing out that I couldn't test it. Changes look correct though.

Reply:
> I'm not recommending removing the method in this PR

Although we could since we're here testing it 😅 

The `getUserExistsError()` check all depends on the signup framework passing in the step data as a prop. Which it does here:
https://github.com/Automattic/wp-calypso/blob/73e33a1213bd9e4dd34c379a086f2c18ebbb68c4/client/signup/steps/user/index.jsx#L447

The signup framework is the only use that does this, and it seems like a very strange violation of the component boundary having it deal with signup framework internals like this.

Reply:
Thanks for taking the time to take a look thoroughly!

Agreed, this part of the code definitely needs some investigation and cleanup of the dead unreachable areas. 

I think it's material for another PR though - I intentionally didn't clean it up to keep the PR focused on one task at a time.